src/mmw/js/src/modeling/models.js

```diff
@@ -825,14 +825,17 @@ var SubbasinDetailModel = Backbone.Model.extend({
 
     fetchCatchmentsIfNeeded: function(comids) {
         var catchments = this.get('catchments');
-        if (!catchments.isEmpty() || this.fetchCatchmentsPromise) { return; }
+        if (!catchments.isEmpty() || this.fetchCatchmentsPromise) {
+            return $.when();
```

Comment:
Sorry, I missed this earlier, but if `this.fetchCatchmentsPromise` exists, we should probably return that instead of `$.when()`:

```js
if (!catchments.isEmpty() || this.fetchCatchmentsPromise) {
  return this.fetchCatchmentsPromise || $.when();
}
```

Reply:
Oh, that's my bad. I'll fix before merging.